src/tests/dsETH.test.ts

```diff
@@ -1,55 +1,604 @@
+import { swapExactInput } from './utils/uniswap'
 import {
-  AlchemyProvider,
+  approveErc20,
+  balanceOf,
+  createERC20Contract,
   LocalhostProvider,
   SignerAccount0,
+  wrapETH,
   ZeroExApiSwapQuote,
-} from 'tests/utils'
+} from './utils'
 
-describe('FlashMintZeroEx - Quotes for dsETH', () => {
-  const chainId = 1
-  const zeroExApi = ZeroExApiSwapQuote
+const provider = LocalhostProvider
 
+const dsETH = {
+  address: EthereumDiversifiedStakingIndex.address!,
+  decimals: 18,
+  symbol: EthereumDiversifiedStakingIndex.symbol,
+}
+
+const ETH = {
+  address: '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE',
+  decimals: 18,
+  symbol: 'ETH',
+}
+const RETH = {
+  address: '0xae78736Cd615f374D3085123A210448E74Fc6393',
+  decimals: 18,
+  symbol: 'rETH',
+}
+const SETH2 = {
+  address: sETH2.address!,
+  decimals: 18,
+  symbol: sETH2.symbol,
+}
+const STETH = {
+  address: stETH.address!,
+  decimals: 18,
+  symbol: stETH.symbol,
+}
+const WETH9 = {
+  address: WETH.address!,
+  decimals: 18,
+  symbol: WETH.symbol,
+}
+const WSETH2 = {
+  address: wsETH2.address!,
+  decimals: 18,
+  symbol: wsETH2.symbol,
+}
+const WSTETH = {
+  address: wstETH.address!,
+  decimals: 18,
+  symbol: wstETH.symbol,
+}
+
+describe('FlashMintZeroEx - dsETH', () => {
   beforeEach((): void => {
     jest.setTimeout(10000000)
   })
 
-  test('returns a quote for minting', async () => {
-    const isMinting = true
-    // const setToken = '0x683Bad7EB64cE3Ec13eCeC3bc7583FCBCB75A9F7'
-    // const setTokenSymbol = 'dsETH'
-    // FIXME:
-    const setToken = '0x72e364F2ABdC788b7E918bc238B21f109Cd634D7'
-    const setTokenSymbol = MetaverseIndex.symbol
-    const setTokenAmount = wei('1')
-    const inputToken = {
-      address: '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE',
-      decimals: 18,
-      symbol: 'ETH',
-    }
-    const outputToken = {
-      address: setToken,
-      decimals: 18,
-      symbol: setTokenSymbol,
-    }
-    const quote = await getFlashMintZeroExQuote(
+  test('minting with ETH', async () => {
+    const inputToken = ETH
+    const outputToken = dsETH
+    const indexTokenAmount = wei('0.1')
+
+    await mint(outputToken, indexTokenAmount)
+  })
+
+  test('redeeming to ETH', async () => {
+    const inputToken = dsETH
+    const outputToken = ETH
+    const indexTokenAmount = wei('0.1')
+
+    await redeem(inputToken, indexTokenAmount)
+  })
+
+  test('minting with WETH', async () => {
+    const inputToken = WETH9
+    const outputToken = dsETH
+    const indexTokenAmount = wei('1')
+
+    const signer = SignerAccount0
+    await wrapETH(wei(2), signer)
+
+    await mintERC20(inputToken, outputToken, indexTokenAmount, 0.5, signer)
+  })
+  test('redeeming to WETH', async () => {
+    const inputToken = dsETH
+    const outputToken = WETH9
+    const indexTokenAmount = wei('1')
+
+    await redeemERC20(inputToken, outputToken, indexTokenAmount)
+  })
+
+  // FIXME: deposit into pool won't work because it reached max capacity
```

Comment:
The comment says it all. Also, tried getting `rETH` with `swapExactInput` utility function but there seem to be some issues with that Uniswap pool (at least with the forked env).